sw/airborne/modules/guidance/gvf/trajectories/gvf_ellipse.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2016  Hector Garcia de Marina
+ *
+ * This file is part of paparazzi.
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ *
+ */
+
+/** \file gvf_ellipse.c
+ *
+ *  Guidance algorithm based on vector fields
+ *  2D Ellipse trajectory
+ */
+
+
+#include "subsystems/navigation/common_nav.h"
+#include "gvf_ellipse.h"
+
+#ifndef GVF_ELLIPSE_A
+#define GVF_ELLIPSE_A 80
+#endif
+
+#ifndef GVF_ELLIPSE_B
+#define GVF_ELLIPSE_B 80
+#endif
+
+#ifndef GVF_ELLIPSE_ALPHA
+#define GVF_ELLIPSE_ALPHA 0
+#endif
+
+gvf_ell_par gvf_ellipse_par = {GVF_ELLIPSE_A, GVF_ELLIPSE_B, GVF_ELLIPSE_ALPHA};
+
+void gvf_ellipse_info(float *phi, struct gvf_grad *grad,
+                      struct gvf_Hess *hess)
+{
+
+  struct EnuCoor_f *p = stateGetPositionEnu_f();
+  float px = p->x;
+  float py = p->y;
+  float wx = gvf_trajectory.p[0];
+  float wy = gvf_trajectory.p[1];
+  float a = gvf_trajectory.p[2];
+  float b = gvf_trajectory.p[3];
+  float alpha = gvf_trajectory.p[4];
+
+  // Phi(x,y)
+  float xel = (px - wx) * cosf(alpha) - (py - wy) * sinf(alpha);
```

Comment:
haha yep. No worries, it does not take any effort to make this change. I will do it within tomorrow :P.